### ISSUE TITLE
Only last stream's progress bar removed 

### ISSUE THREAD

**Issue by gnovis** [Cognexa/cxflow#142]
In my opinion, all progress bars should be removed or all progress bars should stay on a screen.
![bp](https://user-images.githubusercontent.com/4740230/30782515-fe202972-a133-11e7-9cf1-09028c40c19c.png)
The screen from training with `train` and `test` stream.

**Comment by blazekadam** [comment]
Well, it is in fact quite messy as any hooks may interfere with `ShowProgress` and even the hook order can come into play.

I tried to refactor the code so that it is more readable. These print artifacts are the side-effects.

As you probably noticed, the hook no longer removes the lane after the stream is expected to end. The idea is that it may be unreliable. What would you suggest?

**Comment by gnovis** [comment]
Removing the following two lines will fix this. I don't see the problem with it, is there any?
```python
 98             if self._current_stream_name is not None:
 99                 print()
```

**Comment by blazekadam** [comment]
Closed with #154
